Author: aszlig <aszlig@nix.build>
Date:   Sat Aug 4 23:09:37 2018 +0200

    Add test for imageproxy URLs in Markdown
    
    This is a small Selenium/Nightwatch test which ensures that image URLs
    in markdown are properly appended to the image proxy URL.
    
    Note that we don't actually verify whether the proxy fetches the image
    correctly, only whether the Habitica client encodes the URL correctly.
    
    Testing the proxy itself is done in a NixOS VM test.
    
    Signed-off-by: aszlig <aszlig@nix.build>
    Filename: test-imageproxy-in-markdown.patch

diff --git a/test/client/e2e/specs/test.js b/test/client/e2e/specs/test.js
index f439644760..8898b7c0d4 100644
--- a/test/client/e2e/specs/test.js
+++ b/test/client/e2e/specs/test.js
@@ -15,4 +15,32 @@ module.exports = {
       .assert.containsText('h1', 'Motivate yourself to achieve your goals.')
       .end();
   },
+  'images in markdown use proxy url' (browser) {
+    const baseUrl = browser.globals.devServerURL;
+
+    browser
+      .url(`${baseUrl}/register`)
+      .waitForElementVisible('#app', 50000)
+      .setValue('#usernameInput', 'foo')
+      .setValue('#emailInput', 'foo@example.org')
+      .setValue('#passwordInput', 'bar')
+      .setValue('#confirmPasswordInput', 'bar')
+      .useXpath()
+      .click('//*[text()="Join Habitica"]')
+      .useCss()
+      .waitForElementVisible('.welcome-section', 50000)
+      .click('#avatar-modal .btn-primary')
+      .waitForElementVisible('.avatar-section.page-2', 50000)
+      .click('.next')
+      .click('.next')
+      .waitForElementVisible('.introjs-donebutton', 50000)
+      .click('.introjs-donebutton')
+      .setValue('textarea.quick-add', [
+        'Test ![image](http://example.org/image.png)',
+        [browser.Keys.ENTER],
+      ])
+      .waitForElementVisible('.markdown-img', 50000)
+      .assert.attributeEquals('.markdown-img', 'src',
+        `${baseUrl}/imageproxy/aHR0cDovL2V4YW1wbGUub3JnL2ltYWdlLnBuZw`);
+  },
 };
